src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -1732,3 +1558,46 @@ _mongoc_topology_bypass_cooldown (mongoc_topology_t *topology)
    BSON_ASSERT (topology->single_threaded);
    topology->scanner->bypass_cooldown = true;
 }
+
+/* Called from application threads
+ * Caller must hold topology lock.
+ * Locks topology description mutex to copy out server description errors.
+ * CLEANUP: this has nothing to do with background monitoring. Move to
+ * mongoc-topology.c.
+ * For single-threaded monitoring, the topology scanner may include errors for
+ * servers that were removed from the topology.
+ */
+static void
+_topology_collect_errors (mongoc_topology_t *topology, bson_error_t *error_out)
```

Comment:
The single-threaded mongoc_topology_scanner_t keeps the last seen error per server scanned. This could include an error received on a server that was removed from the topology if it had been scanned during the full blocking scan.
But in multi-threaded, servers are scanned independently, there's no clear way to track errors of servers that were removed (how long would we keep those errors around for?).

Instead, take errors from the server descriptions in the topology. This aligns with the server selection spec's recommendation of how to return errors:
https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#server-selection-errors